framework/source/class/qx/ui/core/queue/Appearance.js

```diff
@@ -88,6 +94,7 @@ qx.Class.define("qx.ui.core.queue.Appearance",
         // Order is important to allow the same widget to be re-queued directly
         obj = queue[i];
         queue.splice(i, 1);
+        delete this.__lookup[obj.$$hash]
```

Comment:
possibly, provided that the call to obj.syncAppearance() on line 101 does not try to put things back on the queue.  As the original code edits the array rather than just resetting it, this follows the same policy and I felt it was safer.

(actually Visibility.js does exactly what you suggest)